components/event-publisher-proxy/pkg/handler/handler.go

```diff
@@ -317,7 +318,12 @@ func (h *Handler) sendEventAndRecordMetrics(ctx context.Context, event *cev2even
 		h.collector.RecordBackendError()
 		return nil, err
 	}
-	h.collector.RecordEventType(event.Type(), event.Source(), result.HTTPStatus())
+	originalEventType, ok := event.Extensions()[builder.OriginalTypeHeaderName].(string)
+	if !ok {
+		h.namedLogger().With().Errorw("failed to convert event original event type extension value to string", builder.OriginalTypeHeaderName, event.Extensions()[builder.OriginalTypeHeaderName])
+		return nil, fmt.Errorf("failed to convert event extension %s value %d to string", builder.OriginalTypeHeaderName, event.Extensions()[builder.OriginalTypeHeaderName])
```

Comment:
1. please also check if its not empty value.
2. Instead of returning error, why dont we log a warning and use the prefixed type like:
```
if !ok && originalEventType == "" {
    h.namedLogger().With().Warningw("failed to convert event original event type extension value to string", builder.OriginalTypeHeaderName, event.Extensions()[builder.OriginalTypeHeaderName])

    originalEventType = event.Type()
}
```

Reply:
and also please add unit tests.

Reply:
done